staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -248,7 +250,14 @@ func (n typeNode) lookupField(jsonName string) *childNode {
 	return nil
 }
 
-func (n typeNode) hasValidations(nodes map[*types.Type]*typeNode) bool {
+func (n typeNode) hasValidations(nodes map[*types.Type]*typeNode) (result bool) {
+	if n.cachedHasValidations != nil {
+		return *n.cachedHasValidations
+	}
+	defer func() {
+		n.cachedHasValidations = &result
```

Comment:
Does this work?  In the places you retrun a literal, is Go actually setting that as the value of `result` ?  I have never seen that pattern, if so.

Reply:
Yeah, it's valid go.  Defers are evaluated "after" the function is evaluated and returns are assigned.. I had to check the language spec and validate it in the debugger. If it's too magical I can drop it, but it seems like a neat memoize recipe so I figured I'd leverage it.